### ISSUE TITLE
Unzip benchmark

### ISSUE THREAD

**Issue by shinnn** [kevva/decompress-unzip#3]
I created a benchmark to see which unzip module is the fastest.

## modules

* [adm-zip](https://github.com/cthackers/adm-zip) (currently this project uses)
* [decompress-zip](https://github.com/bower/decompress-zip)
* [extract-zip](https://github.com/maxogden/extract-zip)
* [node-unzip](https://github.com/EvanOxfeld/node-unzip)

## benchmark script

(Using [matcha](https://github.com/logicalparadox/matcha))

```js
'use strict';

var fs = require('fs');

var rimraf = require('rimraf');

var AdmZip = require('adm-zip');
var DecompressZip = require('decompress-zip');
var extractZip = require('extract-zip');
var unzip = require('unzip');

var FIXTURE_PATH = 'fixture.zip'; // path to a ZIP file

suite('Extract all files from a ZIP archive', function() {
  set('iterations', 10);

  before(rimraf.bind(null, 'tmp'));

  bench('adm-zip', function() {
    new AdmZip(FIXTURE_PATH).extractAllTo('tmp/adm-zip', true);
  });

  bench('decompress-zip', function(next) {
    new DecompressZip(FIXTURE_PATH)
    .on('extract', next)
    .on('error', next)
    .extract({path: 'tmp/decompress-zip'});
  });

  bench('extract-zip', function(next) {
    extractZip(FIXTURE_PATH, {dir: 'tmp/extract-zip'}, next);
  });

  bench('node-unzip', function(next) {
    var extract = new unzip.Extract({path: 'tmp/unzip/'})
    .on('finish', next)
    .on('error', next);

    fs.createReadStream(FIXTURE_PATH).pipe(extract);
  });
});
```

### result

```sh
                      Extract all files from a ZIP archive
              43 op/s » adm-zip
             112 op/s » decompress-zip
              70 op/s » extract-zip
              49 op/s » node-unzip

```

I think we should replace adm-zip with another module.

**Comment by kevva** [comment]
Too bad `decompress-zip` doesn't support listing the files contents and their attributes on the `list` event.

**Comment by kevva** [comment]
Fixed in https://github.com/kevva/decompress-unzip/commit/253d4f1afafada9b052e784f4bc011bf8d7d046d. @shinnn, do you think it's safe to release this is a minor? Or how much testing have you done with various zip files?

**Comment by shinnn** [comment]
At least [pngcrush-bin](https://github.com/imagemin/pngcrush-bin) works well. I'm trying to test other zip files...

**Comment by kevva** [comment]
I've added you to the repo. Feel free to publish when you feel confident it's working.

**Comment by kevva** [comment]
@shinnn, https://github.com/sindresorhus/flow-bin/issues/7.

**Comment by shinnn** [comment]
Yes, I already know `write after EOF` problem.

**Comment by shinnn** [comment]
I released v2.1.1. [flow-bin](https://github.com/sindresorhus/flow-bin) passes the test with v2.1.1.

**Comment by kevva** [comment]
Sweet.
